### PR TITLE
Add support for Python 3.12

### PR DESCRIPTION
**Pull request by troibe**
All python requirements now support Python 3.12

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@
 # lines are added as extras under a group called "<tag>".
 
 numpy >= 1.19
-aiohttp >= 3.7.4.post0
+aiohttp >= 3.9.0b0
```
Comment (gadfort) [inline requirements.txt:10]: we should check against the rc0 they have

**Review by gadfort — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 2]
```diff
@@ -20,7 +20,8 @@ packaging >= 21.3
 psutil >= 5.8.0
 Pillow >= 8.4.0
 GitPython >= 3.1.0
-sc-leflib == 0.1.0
+sc-leflib >= 0.2.0
+pyarrow >= 14.0.0
```
[the remark this answers is not in the document]
Comment (troibe) [reply requirements.txt:24]: Let me check streamlit. I wasn't sure initially where the dependency coming from.

[the remark this answers is not in the document]
Comment (troibe) [reply requirements.txt:24]: Actually that isn't even required. Pyarrow 14.0.0 seems to get used automatically now.

File: requirements.txt (hunk 1 above)
Comment (troibe) [reply requirements.txt:10]: Sure they have that too.

**Comment by troibe** [comment]
`Altair` seems to be fine to remove as well.
Successful wheels run here: https://github.com/siliconcompiler/siliconcompiler/actions/runs/6880430018

**Comment by gadfort** [comment]
@troibe according to the aiohttp repo the 3.9.0 release should happen this weekend assuming no new issues are found, so I think it makes sense to wait until the release is available

**Comment by troibe** [comment]
sounds good

**Comment by gadfort** [comment]
For reference: https://github.com/aio-libs/aiohttp/issues/7675#issuecomment-1810330313

**Review by gadfort — APPROVED** [review]

**Pull request merged**